src/Draco.Compiler/Internal/Documentation/DocumentationSection.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Draco.Compiler.Internal.Documentation;
+
+/// <summary>
+/// Represents a section of the documentation.
+/// </summary>
+internal sealed class DocumentationSection
+{
+    public string? Name { get; }
```

Comment:
If the section kind was passed in and is not other, this could still be filled out. But can stay nullable